src/frontend/js/components/PaymentButton/index.tsx

```diff
@@ -102,12 +102,10 @@ const PaymentButton = ({ billingAddress, creditCard, onSuccess }: PaymentButtonP
   const [payment, setPayment] = useState<PaymentInfo>();
   const [state, setState] = useState<ComponentStates>(ComponentStates.IDLE);
   const [error, setError] = useState<PaymentErrorMessageId>(PaymentErrorMessageId.ERROR_DEFAULT);
-  const paymentId = useMemo(() => {
-    if (payment && payment.hasOwnProperty('payment_id')) {
-      return (payment as Exclude<Joanie.Payment, Joanie.LyraPayment>).payment_id;
-    }
-    return undefined;
-  }, [payment]);
+  const hasPaymentId = (p: Maybe<Joanie.Payment>): p is Extract<Joanie.Payment, PaymentWithId> => {
```

Comment:
i don't understand why you need to use `Extract`, `p is PaymentWithId` would work the same.

i thinks the standard naming for typeguard is :
`is[MyType]`, so here: `isPaymentWithId`

Reply:
Yup. But `MyType` should be a type which complies with the type of the parameter.

So a `Payment` object cannot be of type `PaymentWithId`. That's why I use Extract. I extract from `Payment` type only unioned types which extends of `PaymentWithId`.